client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -0,0 +1,578 @@
+		// still running (even with the dot below the dock icon).
+		obj := cocoa.NSStatusBar_System().StatusItemWithLength(cocoa.NSVariableStatusItemLength)
+		obj.Retain()
+		obj.Button().SetImage(cocoa.NSImage_InitWithData(core.NSData_WithBytes(SymbolBWIcon, uint64(len(SymbolBWIcon)))))
+		obj.Button().Image().SetSize(core.Size(18, 18))
+		obj.Button().SetToolTip("Self-custodial multi-wallet")
+
+		runningItem := cocoa.NSMenuItem_New()
+		runningItem.SetTitle("Dex Client is running")
+		runningItem.SetEnabled(false)
+
+		itemQuit := cocoa.NSMenuItem_Init("Force Quit", objc.Sel("terminate:"), "q")
+		itemQuit.SetToolTip("Force DEX client to close")
+
+		menu := cocoa.NSMenu_New()
+		menu.AddItem(runningItem)
+		menu.AddItem(itemQuit)
+		obj.SetMenu(menu)
+
+		// Hide the application until it is ready to be shown when we receive
+		// the "applicationDidFinishLaunching" notification below. This also
+		// allows us to ensure the menu bar is redrawn.
+		app.TryToPerform_with_(objc.Sel("hide:"), nil)
+	})
+
+	// MacOS will always send the "applicationDidFinishLaunching" event when an
+	// application has finished launching. See:
+	// https://developer.apple.com/documentation/appkit/nsapplicationdidfinishlaunchingnotification?language=objc
+	addMethodToDelegate("applicationDidFinishLaunching:", func(_, notification objc.Object) {
+		// Unhide the app on the main thread after it has finished launching we
+		// need to give this priority before creating the window to ensure the
+		// window is immediately visible when it's created. This also has the
+		// side effect of redrawing the menu bar which will be unresponsive
+		// until it is redrawn.
+		core.Dispatch(func() {
+			app.TryToPerform_with_(objc.Sel("unhide:"), nil)
+		})
+
+		createNewWebView()
+	})
+
+	app.SetActivationPolicy(cocoa.NSApplicationActivationPolicyRegular)
+	app.ActivateIgnoringOtherApps(false)
+	app.SetDelegate(cocoa.DefaultDelegate)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-appCtx.Done():
+				cocoa.NSApp().Terminate()
+				return
+			}
+		}
+	}()
+
+	app.Run() // blocks until app.Terminate() is executed and wil not continue execution of any defer statements in the main thread.
+	return nil
+}
+
+// createMainMenuItems creates the main menu items for the app menu.
+func createMainMenuItems() []cocoa.NSMenu {
+	// Create the App menu.
+	appMenu := cocoa.NSMenu_Init(macOSAppTitle)
+
+	// Add the menu items.
+	hideMenuItem := cocoa.NSMenuItem_Init("Hide "+macOSAppTitle, objc.Sel("hide:"), "h")
+	appMenu.AddItem(hideMenuItem)
+
+	hideOthersMenuItem := cocoa.NSMenuItem_Init("Hide Others", objc.Sel("hideOtherApplications:"), "")
+	appMenu.AddItem(hideOthersMenuItem)
+
+	showAllMenuItem := cocoa.NSMenuItem_Init("Show All", objc.Sel("unhideAllApplications:"), "")
+	appMenu.AddItem(showAllMenuItem)
+
+	appMenu.AddItem(cocoa.NSMenuItem_Separator())
+
+	quitMenuItem := cocoa.NSMenuItem_Init("Quit "+macOSAppTitle, objc.Sel("terminate:"), "q")
```

Comment:
Yup, the terminology used for terminating an app in the Main menu is always `Quite App Name` for most macOS apps. The `itemQuit` item in the system menu bar uses `Force Quit` since that's not the usual place to quit the app, but for the sake of harmony, we can use Quit everywhere.